test/puppetlabs/puppetdb/testutils/log.clj

```diff
@@ -0,0 +1,164 @@
+(ns puppetlabs.puppetdb.testutils.log
+  (:require
+   [puppetlabs.kitchensink.core :as kitchensink]
+   [puppetlabs.puppetdb.testutils :refer [temp-file]]
+   [me.raynes.fs :as fs])
+  (:import
+   [ch.qos.logback.core Appender FileAppender]
+   [ch.qos.logback.core.spi LifeCycle]
+   [ch.qos.logback.classic Level Logger]
+   [ch.qos.logback.classic.encoder PatternLayoutEncoder]
+   [org.slf4j LoggerFactory]))
+
+(defn- find-logger [id]
+  (.getLogger (LoggerFactory/getILoggerFactory)
+              (if (class? id) id (str id))))
+
+(defmacro with-log-level
+  "Sets the (logback) log level for the logger specified by logger-id
+  during the execution of body.  If logger-id is not a class, it is
+  converted via str, and the level must be a clojure.tools.logging
+  key, i.e. :info, :error, etc."
+  [logger-id level & body]
+  ;; Specify the root logger via org.slf4j.Logger/ROOT_LOGGER_NAME.
+  ;; Assumes use of logback (i.e. logger supports Levels).
+  `(let [logger# (#'puppetlabs.puppetdb.testutils.log/find-logger ~logger-id)
```

Comment:
Yes, it's so the expansion will still work.  I'm pretty sure I verified that, and it's also needed iirc in (for example) Guile.